components/proxy/src/main/java/com/hotels/styx/spi/ObjectFactories.java

```diff
@@ -73,27 +64,27 @@ public ObjectFactory(@JsonProperty("class") String factoryClass,
                 T instance = type.cast(aClass.newInstance());
                 instances.add(instance);
             } catch (ClassNotFoundException e) {
-                LOG.debug("", e);
+                LOGGER.debug("", e);
             } catch (InstantiationException | IllegalAccessException e) {
-                LOG.error("error loading class={}", className);
+                LOGGER.error("error loading class={}", className);
                 return Optional.empty();
             }
         }
 
         int size = instances.size();
         if (size == 0) {
-            LOG.error("no class={} is found in the specified classpath={}", className, pathOptional.orElse(Paths.get("")));
+            LOGGER.error("no class={} is found in the specified classpath={}", className, pathOptional.orElse(Paths.get("")));
```

Comment:
Hmm.. I think the new `SpiExtension` class requires `classPath` to be present. I believe it will fail even before it gets this far. This is because I separated the logic of class loading away from the configuration classes. Loading happens only *after* Yaml configuration has been fully deserialised.